### PR TITLE
DAOS-6334 tests: use assert_rc_equal for unit tests

### PR DESCRIPTION
**Pull request by marcelarosalesj**
Signed-off-by: Marcela Rosales <marcela.a.rosales.jimenez@intel.com>

### CODE REVIEW

File: src/vea/tests/vea_ut.c [hunk 1]
```diff
@@ -227,14 +228,14 @@ ut_reserve(void **state)
 
 	/* Verify transient is allocated */
 	rc = vea_verify_alloc(args->vua_vsi, true, off_a, blk_cnt);
-	assert_int_equal(rc, 0);
+	assert_rc_equal(rc, 0);
 	/* Verify persistent is not allocated */
 	rc = vea_verify_alloc(args->vua_vsi, false, off_a, blk_cnt);
-	assert_int_equal(rc, 1);
+	assert_rc_equal(rc, 1);
```
Comment (jolivier23) [inline src/vea/tests/vea_ut.c:234]: the ones that expect 1 are a little weird.  At the least, they need your patch to update the macro.

**Review by jolivier23 — APPROVED** [review]

File: src/vea/tests/vea_ut.c (hunk 1 above)
Comment (kjacque) [reply src/vea/tests/vea_ut.c:234]: Agreed, this is weird. And in certain cases (rc doesn't map to a DER value) the assert comparing the strings in the macro just won't detect the failure.

**Review by kjacque — APPROVED** [review]
Okay with landing, but the macro really needs to be updated.

**Comment by marcelarosalesj** [comment]
> Okay with landing, but the macro really needs to be updated.

You're right, this PR https://github.com/daos-stack/daos/pull/4739 will consider those cases

**Pull request merged**